conans/test/functional/layout/test_in_cache.py

```diff
@@ -147,6 +147,7 @@ def test_cache_in_layout(conanfile):
     assert "Package_ID: {}".format(package_id) in client.out
 
 
+@pytest.mark.xfail(reason="The conan-source command do not handle yet the layout, to do in other PR")
```

Comment:
@lasote the ``conan source`` case has not been addressed in this PR, it will be later in a separate one, better to keep this one focused on the profile issue.